diff --git a/cmake/base-config-ix.cmake b/cmake/base-config-ix.cmake
index 2a44d830b..78ac1114c 100644
--- a/cmake/base-config-ix.cmake
+++ b/cmake/base-config-ix.cmake
@@ -89,16 +89,16 @@ else(LLVM_ENABLE_PER_TARGET_RUNTIME_DIR)
 endif()
 
 if(APPLE)
-  # On Darwin if /usr/include doesn't exist, the user probably has Xcode but not
-  # the command line tools. If this is the case, we need to find the OS X
-  # sysroot to pass to clang.
-  if(NOT EXISTS /usr/include)
-    execute_process(COMMAND xcodebuild -version -sdk macosx Path
-       OUTPUT_VARIABLE OSX_SYSROOT
-       ERROR_QUIET
-       OUTPUT_STRIP_TRAILING_WHITESPACE)
-    set(OSX_SYSROOT_FLAG "-isysroot${OSX_SYSROOT}")
-  endif()
+  # # On Darwin if /usr/include doesn't exist, the user probably has Xcode but not
+  # # the command line tools. If this is the case, we need to find the OS X
+  # # sysroot to pass to clang.
+  # if(NOT EXISTS /usr/include)
+  #   execute_process(COMMAND xcodebuild -version -sdk macosx Path
+  #      OUTPUT_VARIABLE OSX_SYSROOT
+  #      ERROR_QUIET
+  #      OUTPUT_STRIP_TRAILING_WHITESPACE)
+  #   set(OSX_SYSROOT_FLAG "-isysroot${OSX_SYSROOT}")
+  # endif()
 
   option(COMPILER_RT_ENABLE_IOS "Enable building for iOS" On)
   option(COMPILER_RT_ENABLE_WATCHOS "Enable building for watchOS - Experimental" Off)
